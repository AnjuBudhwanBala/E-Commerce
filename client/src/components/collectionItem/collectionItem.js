import React, { useCallback } from 'react';

import { useDispatch } from 'react-redux';
import { addItem } from '../../redux/cart/cart.actions';
import {
  CollectionItemContainer,
  CollectionFooterContainer,
  AddButton,
  BackgroundImage,
  NameContainer,
  PriceContainer
} from './collectionItem.styles';

const CollectionItem = ({ item }) => {
  const { imageUrl, price, name } = item;

  const dispatch = useDispatch(() => dispatch);

  const addCartItem = useCallback(
    item => {
      dispatch(addItem(item));
    },
    [dispatch]
  );

  return (
    <CollectionItemContainer>
      <BackgroundImage className="image" imageUrl={imageUrl} />
      <CollectionFooterContainer>
        <NameContainer>{name}</NameContainer>
        <PriceContainer>{price}</PriceContainer>
      </CollectionFooterContainer>
      <AddButton inverted onClick={() => addCartItem(item)}>
        Add to Cart
      </AddButton>
    </CollectionItemContainer>
  );
};

export default CollectionItem;
