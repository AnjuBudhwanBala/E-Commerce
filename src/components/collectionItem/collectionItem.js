import React, { useCallback } from 'react';
import './collectionItem.styles.scss';
import CustomButton from '../customButton/customButton';
import { useDispatch } from 'react-redux';
import { addItem } from '../../redux/cart/cart.actions';

const CollectionItem = ({ item }) => {
  const { imageUrl, price, name } = item;

  const dispatch = useDispatch(() => dispatch);

  const addCartItem = useCallback(
    item => {
      dispatch(addItem(item));
    },
    [dispatch]
  );

  return (
    <div className="collectionItem">
      <div className="image" style={{ backgroundImage: `url(${imageUrl})` }} />
      <div className="collection-footer">
        <span className="name">{name}</span>
        <span className="price">{price}</span>
      </div>
      <CustomButton inverted onClick={() => addCartItem(item)}>
        Add to Cart
      </CustomButton>
    </div>
  );
};

export default CollectionItem;
