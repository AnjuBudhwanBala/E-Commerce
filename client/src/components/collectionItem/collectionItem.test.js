import React from 'react';
import { shallow } from 'enzyme';
import { CollectionItem } from './collectionItem';

describe('CollectionItem Component', () => {
  let wrapper;
  let mockAddItem;
  const imageUrl = 'www.testImage.com';
  const mockName = 'black hat';
  const mockPrice = 10;

  beforeEach(() => {
    mockAddItem = jest.fn();
    const mockProps = {
      item: {
        name: mockName,
        price: 10,
        imageUrl
      },
      addItem: mockAddItem
    };
    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it('should render CollectionItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addItem when AddButton clicked', () => {
    wrapper.find('AddButton').simulate('click');
    expect(mockAddItem).toHaveBeenCalled();
  });

  it('should render ImageUrl as a prop on BackgroundImage', () => {
    expect(wrapper.find('BackgroundImage').prop('imageUrl')).toBe(imageUrl);
  });

  it('should render name prop in NameContainer', () => {
    expect(wrapper.find('NameContainer').text()).toBe(mockName);
  });
  it('should render price prop in PriceContainer', () => {
    const price = parseInt(wrapper.find('PriceContainer').text());
    expect(price).toBe(mockPrice);
  });
});
