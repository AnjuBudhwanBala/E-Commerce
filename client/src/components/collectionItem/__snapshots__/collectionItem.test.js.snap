// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem Component should render CollectionItem component 1`] = `ShallowWrapper {}`;
